log/field_encoders.go

```diff
@@ -0,0 +1,49 @@
+package log
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+)
+
+// FieldEncoder describes a way to encode fields.
+type FieldEncoder func(io.Writer, Field) error
```

Comment:
Instead of having an `EncodeMany` function, would it make sense to make the `FieldEncoder` type variadic?
```go
// FieldEncoder describes a way to encode fields.
type FieldEncoder func(io.Writer, ...Field) error
```